src/libaktualizr/storage/sqlstorage_base.cc

```diff
@@ -61,6 +63,48 @@ std::string SQLStorageBase::getTableSchemaFromDb(const std::string& tablename) {
   return schema.value() + ";";
 }
 
+bool SQLStorageBase::dbMigrateForward(int version_from) {
+  SQLite3Guard db = dbConnection();
+  for (int32_t k = version_from + 1; k <= current_schema_version_; k++) {
+    if (db.exec(schema_migrations_.at(static_cast<size_t>(k)), nullptr, nullptr) != SQLITE_OK) {
+      LOG_ERROR << "Can't migrate db from version " << (k - 1) << " to version " << k << ": " << db.errmsg();
+      return false;
+    }
+    if (schema_rollback_migrations_.count(k) != 0) {
+      auto statement =
+          db.prepareStatement("INSERT INTO migrations VALUES (?,?);", k, schema_rollback_migrations_.at(k));
+      if (statement.step() != SQLITE_DONE) {
+        LOG_ERROR << "Can't insert rollback migration script: " << db.errmsg();
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
+bool SQLStorageBase::dbMigrateBackward(int version_from) {
+  SQLite3Guard db = dbConnection();
+  for (int ver = version_from; ver > current_schema_version_; --ver) {
+    auto statement = db.prepareStatement("SELECT migration FROM migrations WHERE version_from=?;", ver);
+    if (statement.step() != SQLITE_ROW) {
+      LOG_ERROR << "Can't extract migration script: " << db.errmsg();
+      return false;
+    }
+    auto migration = statement.get_result_col_str(0);
+    statement.step();
+    if (db.exec(*migration, nullptr, nullptr) != SQLITE_OK) {
+      LOG_ERROR << "Can't migrate db from version " << (ver) << " to version " << ver - 1 << ": " << db.errmsg();
+      return false;
+    }
+    if (db.exec(std::string("DELETE FROM migrations WHERE version_from=") + std::to_string(ver) + ";", nullptr,
```

Comment:
Even though it shouldn't hurt, do you think it's actually needed?

Reply:
Why not? We don't need this anymore so this data is garbage.

Reply:
Ok fair enough, I won't fight on this.